mava/wrappers/matrax.py

```diff
@@ -0,0 +1,93 @@
+# Copyright 2022 InstaDeep Ltd. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Tuple
+
+import chex
+import jax.numpy as jnp
+from jumanji import specs
+from jumanji.env import Environment
+from jumanji.types import TimeStep
+from jumanji.wrappers import Wrapper
+
+from mava.types import Observation, State
+
+
+class MatraxWrapper(Wrapper):
+    def __init__(self, env: Environment):
+        super().__init__(env)
+        self._num_agents = self._env.num_agents
+        self.time_limit = self._env.time_limit
+        self._num_actions = self._env.num_actions
+        self.num_obs_features = self._env.observation_spec().agent_obs.shape[-1]
+
+    def modify_timestep_step(self, timestep: TimeStep) -> TimeStep[Observation]:
+        """Modify the timestep for `step` and `reset`."""
+        observation = Observation(
+            agents_view=timestep.observation.agent_obs,
+            action_mask=jnp.ones((self._num_agents, self._num_actions), dtype=bool),
+            step_count=jnp.repeat(timestep.observation.step_count, self._num_agents),
+        )
+        return timestep.replace(
+            observation=observation,
+            reward=timestep.reward * 1.0,
+            discount=timestep.discount,
+        )
+
+    def modify_timestep_reset(self, timestep: TimeStep) -> TimeStep[Observation]:
+        """Modify the timestep for `step` and `reset`."""
+        observation = Observation(
+            agents_view=timestep.observation.agent_obs,
+            action_mask=jnp.ones((self._num_agents, self._num_actions), dtype=bool),
+            step_count=jnp.repeat(timestep.observation.step_count, self._num_agents),
+        )
+        reward = jnp.repeat(timestep.reward, self._num_agents)
```

Comment:
Can you please install the latest version of Matrax and double check that this still runs? I am pretty sure we fixed this so that the reset won't require reward duplication anymore.

Reply:
This should also allow you to only have single `modify_timstep` method that should work for both `step` and `reset`.

Reply:
Fixed--thanks!